### PR TITLE
Porting kafka consumer from segmentio to confluent. Adding SASL auth to playbook and device consumers

### PR DESCRIPTION
**Pull request by Saurabh-Thakre**
# Description

Add SASL auth to playbook and device consumers
Fixes # (issue)

## Type of change

What is it?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Comment by Saurabh-Thakre** [comment]
Figuring out why https://github.com/RedHatInsights/edge-api/blob/588a267dd8de042f620164f7f24a134b362d501b/pkg/common/kafka/kafkaconfigmap.go#L32 why `cfg.KafkaBrokers` is coming up nil. That is the reason the lint is failing with `Required property group.id not set`

**Comment by Saurabh-Thakre** [comment]
> Figuring out why
> 
> https://github.com/RedHatInsights/edge-api/blob/588a267dd8de042f620164f7f24a134b362d501b/pkg/common/kafka/kafkaconfigmap.go#L32
> 
> why `cfg.KafkaBrokers` is coming up nil. That is the reason the lint is failing with `Required property group.id not set`

Setting group.id outside of the if statement.

**Comment by loadtheaccumulator** [comment]
Please add porting consumer from segmentio to confluent to the PR/commit title.
The SASL description alone is confusing when reading the actual code change.

**Comment by loadtheaccumulator** [comment]
@Saurabh-Thakre Please add porting consumer from segmentio to confluent to the PR/commit title.
The SASL description alone is confusing when reading the actual code change.

**Review by loadtheaccumulator — DISMISSED** [review]
@Saurabh-Thakre Please add porting consumer from segmentio to confluent to the PR/commit title.
The SASL description alone is confusing when reading the actual code change.

File: pkg/services/consumers.go [hunk 1]
```diff
@@ -83,97 +88,151 @@ func (s *KafkaConsumerService) ConsumePlaybookDispatcherRuns() error {
 	// Keep as much logic out of this is method as the Kafka Reader is not mockable for unit tests, as per
```
Comment (AaronH88) [inline pkg/services/consumers.go:88]: We can now remove these comments as they where for segmentio

File: pkg/services/consumers.go [hunk 2]
```diff
@@ -83,97 +88,151 @@ func (s *KafkaConsumerService) ConsumePlaybookDispatcherRuns() error {
...
+		switch e := cs.(type) {
+		case *kafka.Message:
+			key := string(e.Key)
+			var eventType string
+			if key == "event_type" {
+				eventType = string(e.Value)
+			}
+
+			if eventType != InventoryEventTypeCreated && eventType != InventoryEventTypeUpdated && eventType != InventoryEventTypeDelete {
+				//log.Debug("Skipping kafka message - Insights Platform Inventory message is not a created and not an updated event type")
```
Comment (AaronH88) [inline pkg/services/consumers.go:155]: Lets remove this commented out code

File: pkg/services/consumers.go [hunk 3]
```diff
@@ -83,97 +88,151 @@ func (s *KafkaConsumerService) ConsumePlaybookDispatcherRuns() error {
...
+				log.Debug("Skipping message - it is not from edge service")
 			}
-		} else {
-			log.Debug("Skipping message - it is not from edge service")
+		case kafka.Error:
+			// terminate the application if all brokers are down.
+			log.WithFields(log.Fields{"code": e.Code(), "error": e}).Error("Exiting ConsumePlaybookDispatcherRuns loop due to Kafka broker issue")
+			if e.Code() == kafka.ErrAllBrokersDown {
+				run = false
+				return errors.New("uh oh, caught an error due to kafka broker issue") // create an error in errors.go
```
Comment (AaronH88) [inline pkg/services/consumers.go:121]: Do you want to create the error as part of this PR?

Comment (Saurabh-Thakre) [reply pkg/services/consumers.go:121]: sure yeah

**Review by mgold1234 — COMMENTED** [review]
@Saurabh-Thakre can you please add unit test for those changes?
and it's fail on fmt, maybe rebase will help?

**Comment by loadtheaccumulator** [comment]
Just a heads up. While Saurabh is out, I'm going to pull this PR to my local and fix the fmt issue.
I'll try to keep it in this PR, but...

**Pull request merged**